e2e/cypress/integration/messaging/collapse_link_spec.js

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+// ***************************************************************
+// - [number] indicates a test step (e.g. 1. Go to a page)
+// - [*] indicates an assertion (e.g. * Check the title)
+// - Use element ID when selecting an element. Create one if none.
+// ***************************************************************
+
+describe('Messaging', () => {
+    before(() => {
+        // # Set the configuration on Link Previews
+        cy.apiUpdateConfig({
+            ServiceSettings: {
+                EnableLinkPreviews: true,
+            },
+        });
+
+        cy.apiLogin('user-1');
+        cy.apiSaveLinkPreviewPreference(JSON.stringify(true));
+
+        cy.apiLogin('sysadmin');
+        cy.apiSaveLinkPreviewPreference(JSON.stringify(true));
```

Comment:
Let's just do this as
```suggestion
        cy.apiSaveLinkPreviewPreference();
```

Also, I missed that this test also requires test for image previews.  And so, please add saving preference for that as well as ``cy.apiSaveImagePreviewPreference();``

Reply:
I have been checking about the Image Previews, and there are a few things I wanted to check with you.
- Image previews you mean posting a link to an image, or uploading an image?
--- In case of uploading, the preview is only shown on message display Standard. Is this the expected behaviour?
--- In both cases there is no remove button, and therefore this test will not check if removing from one user remove it for all users. This also happens for Youtube videos.

Reply:
Not posting a separate image or image link/URL.  The URL in the test steps (https://www.bbc.com/news/uk-wales-45142614) generates a message attachment with image on it.  The collapse/expand of such image is also controlled by the `collapse_previews` preference.
![Screen Shot 2019-11-13 at 9 51 43 PM](https://user-images.githubusercontent.com/5334504/68770225-b914da00-0660-11ea-8ef2-08d235cd16a5.png)